internal/repository/postgresdb/user.go

```diff
@@ -411,21 +428,32 @@ func (us *userStore) GetUserById(ctx context.Context, reqData dto.GetUserByIdReq
 		return
 	}
 
-	if (userList[0].BadgeCreatedAt.Valid && userList[0].BadgeCreatedAt.Int64 >= reqData.QuaterTimeStamp) || !userList[0].BadgeCreatedAt.Valid {
-		user.UserId = userList[0].UserId
-		user.FirstName = userList[0].FirstName
-		user.LastName = userList[0].LastName
-		user.Email = userList[0].Email
-		user.ProfileImgUrl = userList[0].ProfileImgUrl
-		user.Designation = userList[0].Designation
-		user.RewardQuotaBalance = userList[0].RewardQuotaBalance
-		user.GradeId = userList[0].GradeId
-		user.EmployeeId = userList[0].EmployeeId
-		user.TotalPoints = userList[0].TotalPoints.Int64
-		user.Badge = userList[0].Badge.String
-		user.BadgeCreatedAt = userList[0].BadgeCreatedAt.Int64
+	// Ensure we got at least one result from the database
+	if len(userList) == 0 {
+		logger.Errorf(ctx, "user not found, err: %v", err)
+		err = apperrors.InvalidId
+		return
 	}
-	return
+
+	// Log the retrieved user data for debugging
+	logger.Info(ctx, "userlist: ", userList)
```

Comment:
remove this after testing

Reply:
updated to : logger.Debug(ctx, "userlist: ", userList)